### PR TITLE
Add QGreenland monthly median sea ice extent shape files

### PR DESCRIPTION
**Pull request by alexolinhager**
Adds the monthly median sea ice extent (1981–2010) shape files from the QGreenland dataset. The original line segments delineating sea ice extent have been manually connected to create one coherent shape file.

### CODE REVIEW

**Review by cbegeman — COMMENTED** [review]
@alexolinhager Can you provide any other information in the docs that would help users track down the source of this dataset? Maybe a permanent identifier if that exists?

File: geometric_data/seaice/region/April_Historical_Median_Sea_Ice_Extent/region.geojson [hunk 1]
```diff
@@ -0,0 +1,2160 @@
+{
+    "type": "FeatureCollection",
+    "groupName": "enterGroupName",
+    "features": [
+        {
+            "type": "Feature",
+            "properties": {
+                "name": "December Median Sea Ice Extent (1981-2010)",
```
Comment (xylar) [inline geometric_data/seaice/region/April_Historical_Median_Sea_Ice_Extent/region.geojson:8]: It seems like the name got changed to December but the directory is still April.  That shouldn't be possible if you do `gf.split(fc)` to split out the features into the directories in `geometric_data`.  But I think April is probably correct and the change to December was a mistake?

Comment (alexolinhager) [reply geometric_data/seaice/region/April_Historical_Median_Sea_Ice_Extent/region.geojson]: Thanks @xylar. That was just a typo copied from another file, I just pushed a fix for it

**Comment by alexolinhager** [comment]
> @alexolinhager Can you provide any other information in the docs that would help users track down the source of this dataset? Maybe a permanent identifier if that exists?

Good idea, I just added that to the metadata for each file

File: geometric_features/aggregation/seaice/historical_sea_ice.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
+def qgreenland(gf):
```
Comment (matthewhoffman) [inline geometric_features/aggregation/seaice/historical_sea_ice.py:1]: @xylar , is there a coding style guideline for this repo about correspondence between function/class names and file names?

File: geometric_features/aggregation/__init__.py [hunk 3]
```diff
@@ -1,7 +1,7 @@
 from geometric_features.aggregation.ocean import basins, subbasins, \
     antarctic, greenland, ice_shelves, ismip6, arctic as arctic_ocean, \
     transport, arctic_transport, moc
-from geometric_features.aggregation.seaice import arctic as arctic_seaice
+from geometric_features.aggregation.seaice import qgreenland, arctic as arctic_seaice
```
Comment (matthewhoffman) [inline geometric_features/aggregation/__init__.py:4]: For consistency, I suggest importing `qgreenland` with a name to make it clear its for sea ice:
```suggestion
from geometric_features.aggregation.seaice import arctic as arctic_seaice
from geometric_features.aggregation.seaice import qgreenland as qgreenland_seaice
```
(and then adjusting below accordingly)

File: geometric_features/aggregation/seaice/historical_sea_ice.py [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    """
+    Aggregate Greenland continental shelf regions similar to ISMIP6
```
Comment (matthewhoffman) [inline geometric_features/aggregation/seaice/historical_sea_ice.py:3]: update this comment (looks like it was copied from the other PR)

File: geometric_features/geometric_features.py [hunk 5]
```diff
@@ -80,7 +80,7 @@ def read(self, componentName, objectType, featureNames=None, tags=None,
 
         Parameters
         ----------
-        componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean'}
+        componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean','seaice'}
```
Comment (matthewhoffman) [inline geometric_features/geometric_features.py:83]:
```suggestion
        componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean', 'seaice'}
```
(check line length - might be too long)

File: geometric_features/geometric_features.py [hunk 6]
```diff
@@ -173,7 +173,7 @@ def _download_geometric_features(self, componentName, objectType,
...
```
Comment (matthewhoffman) [inline geometric_features/geometric_features.py:176]:
```suggestion
        componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean', 'seaice'}
```

File: geometric_features/geometric_features.py [hunk 7]
```diff
@@ -223,7 +223,7 @@ def _get_feature_names(self, componentName, objectType, featureNames,
...
```
Comment (matthewhoffman) [inline geometric_features/geometric_features.py:226]:
```suggestion
        componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean', 'seaice'}
```

File: geometric_features/geometric_features.py [hunk 8]
```diff
@@ -298,7 +298,7 @@ def _get_file_name(componentName, objectType, featureName):
 
     Parameters
     ----------
-    componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean'}
+    componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean','seaice'}
```
Comment (matthewhoffman) [inline geometric_features/geometric_features.py:301]:
```suggestion
    componentName : {'bedmachine', 'bedmap2', 'iceshelves', 'landice', 'natural_earth', 'ocean', 'seaice'}
```

**Comment by matthewhoffman** [comment]
@alexolinhager , when I add these regions to QGIS and project to a polar stereographic grid, I see seams at 180 degree longitude.  I remember a discussion about this but I can't find it - is this expected?

image

**Review by matthewhoffman — CHANGES_REQUESTED** [review]
@alexolinhager , thanks for putting this together.  The PR looks good and I just have a few small stylistic requests, plus a question about a seam in the region when plotted as polar stereographic.

**Comment by alexolinhager** [comment]
> @alexolinhager , when I add these regions to QGIS and project to a polar stereographic grid, I see seams at 180 degree longitude. I remember a discussion about this but I can't find it - is this expected?
> 
> image

@matthewhoffman, Thanks for looking over these PRs. The seam is expected for these regions and comes from this script: https://github.com/MPAS-Dev/geometric_features/blob/93f71ba01d0722de7bfb1b9ec3e9129b859b8963/geometric_features/feature_collection.py#L319. Without it, geojson shape files crossing the 180 degree longitude or the poles become disjointed.

 @xylar said they look as he would expect, although I think that conversation happened via Slack

**Comment by alexolinhager** [comment]
> > @alexolinhager Can you provide any other information in the docs that would help users track down the source of this dataset? Maybe a permanent identifier if that exists?
> 
> @alexolinhager , can you add the reference to the documentation as well?

No problem, I made this change

**Comment by xylar** [comment]
@alexolinhager, there are some conflicts here.  I can fix them but the cleanest would be if you're okay with rebasing onto `main` and fixing them that way.

**Comment by alexolinhager** [comment]
> @alexolinhager, there are some conflicts here. I can fix them but the cleanest would be if you're okay with rebasing onto `main` and fixing them that way.

No problem, I'll work on that now

**Comment by alexolinhager** [comment]
@xylar, I rebase this PR onto the new main, so it should be ready to merge

**Comment by alexolinhager** [comment]
@xyler, I rebased the branch and pushed a couple other minor tweaks in accordance with Matt's review. Should be ready to merge now

**Review by matthewhoffman — APPROVED** [review]
@alexolinhager , thanks for the additional changes.  Looks good to me.

**Review by xylar — APPROVED** [review]
Looks good to me.

**Comment by xylar** [comment]
@cbegeman, do you have time to approve or request further changes?

**Review by cbegeman — APPROVED** [review]
@alexolinhager All of my suggestions were addressed. Thanks for your hard work on this!

**Pull request merged**